### PR TITLE
#460 LowestOf + unit test

### PR DESCRIPTION
**Pull request by bedward70**
* The `Min` class was restored and was rename to `LowestOf`.
* Unit tests were added (the `LowestOfTest` class).

P.S.: The `Min` class was deleted in the https://github.com/yegor256/cactoos/commit/25dc45d Yegor Bugayenko <yegor256@gmail.com> on 11/27/17 at 4:05 AM.

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#576](https://github.com/yegor256/cactoos/issues/576) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#576](https://github.com/yegor256/cactoos/issues/576) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

File: src/main/java/org/cactoos/scalar/Folded.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
+/**
```
Comment (fabriciofx) [inline src/main/java/org/cactoos/scalar/Folded.java]: @bedward70 Please, look my review of `HighestOf` PR (#575). All comments are aplicable here too.

**Comment by 0crat** [comment]
@fabriciofx[/z](https://www.0crat.com/u/fabriciofx) this job was assigned to you 8 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
The user @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) resigned from [#576](https://github.com/yegor256/cactoos/issues/576), please stop working

**Comment by 0crat** [comment]
This pull request [#576](https://github.com/yegor256/cactoos/issues/576) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C63314D6Z/432233285). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by amihaiemil** [comment]
@bedward70 This PR is really big, and it looks identical to the PR https://github.com/yegor256/cactoos/pull/575, why is that?

**Comment by amihaiemil** [comment]
@bedward70 there should be 1 PR with ``HighestOff`` + unit tests and 1 PR with ``LowestOf`` -- it seems to me you've merged your branches? You don't have to do that, ever, even if you are working on more stuff concurrently. I also don't understand what is it with the "previous steps" and "Step 3". Keep in mind that tasks and PRs should be as atomic and isolated as possible.

**Comment by bedward70** [comment]
@amihaiemil It's splitted #460. 
I assumed that #576 would be merged before  #575.
if #575 is merged then #576 will keeps 2 classes: `LowestOf` and `LowestOfTest`.

**Comment by amihaiemil** [comment]
@bedward70 You cannot make such assumptions. What if that PR never gets merged? Then, the reviewer from this PR (who may not be the same person), has to remember not to merge the common stuff. 

Please, separate the stuff. Remove from this PR the stuff that is also on the other PR. Leave 1 PR with HighestOf, the other with LowestOf :)

**Comment by amihaiemil** [comment]
@bedward70 Also, change their names accordingly. Something simple, like "HighestOf + unit tests" and "LowestOf + unit test". And in their description there should be a short overview of the changes/implementation and a single link to the originating ticket.

**Comment by amihaiemil** [comment]
@bedward70 ping

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by bedward70** [comment]
@amihaiemil I removed the `HighestOf`, removed the 'HighestOfTest' and changed the title and description.

**Comment by amihaiemil** [comment]
@bedward70 much better, thanks :)

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by amihaiemil** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job, as in [§30](http://datum.zerocracy.com/pages/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by ypshenychka** [comment]
@bedward70 According to our [QA Rules](http://datum.zerocracy.com/pages/policy.html#42):
>Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

Please provide a link to original issue in PR's description.

**Comment by amihaiemil** [comment]
@ypshenychka Couldn't we also count the link from the title? The original issue is in the title as well.

**Comment by ypshenychka** [comment]
@fabriciofx According to our [QA Rules](http://datum.zerocracy.com/pages/policy.html#42):
>The code reviewer found at least three problems in the code.

No issues were addressed in this PR while code review. 
Please confirm that you'll try to find at least three problems while future reviews.

**Comment by ypshenychka** [comment]
@amihaiemil It's a plus to have it in title, however the rules require it to be in description as a must.

**Comment by amihaiemil** [comment]
@ypshenychka I confirm (I think you need to correct your comment, it's not addressed to me) -- this PR was reviewed before as well, so I only did last touches :)

**Comment by ypshenychka** [comment]
@amihaiemil OK, I see. My bad. Thanks for pointing out!

**Comment by amihaiemil** [comment]
@bedward70 please don't forget to link the original issue in the PR's description.

**Comment by amihaiemil** [comment]
@bedward70 ping, also don't forget here :)

**Comment by bedward70** [comment]
@amihaiemil Please check. I was added #460 to description. Is it OK?

**Comment by amihaiemil** [comment]
@bedward70 yes, it is ok. These kind of stuff is required by the QA policy. @ypshenychka please give the verdict here, the ticket has been linked in the description :)

**Comment by ypshenychka** [comment]
@bedward70 Thank you.

**Comment by ypshenychka** [comment]
@0crat Quality acceptable

**Comment by 0crat** [comment]
QA review completed: +15 points just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka), total is [+465](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality was "acceptable": +15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+850](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+10564](https://www.0crat.com/u/yegor256)

**Pull request merged**